### ISSUE TITLE
audio_to_melspec_precessor does not return proper MelSpectrogram

### ISSUE THREAD

**Issue by shangeth** [NVIDIA/NeMo#2469]
```python
import torchaudio
from nemo.collections.tts.models.base import Vocoder

audio, sample_rate = torchaudio.load(audio_path) # sample_rate = 22050
vocoder = Vocoder.from_pretrained('tts_melgan')

audio_len = torch.tensor(audio.size(1)/22050).view(1,1)
gen_mel, _ = vocoder.audio_to_melspec_precessor(audio.cuda(), audio_len.cuda())
plt.imshow(gen_mel.cpu().detach().numpy().squeeze())
plt.show()
```

![image](https://user-images.githubusercontent.com/21142903/125405170-0cf73980-e3d5-11eb-8b25-ebd6fe04d4e8.png)

This code returns a blank(constant) spectrogram.
How do I get the Mel spectrogram for vocoder training/inference?

**Comment by shangeth** [comment]
Hi @titu1994 ,
Any idea about this?. I saw that you contributed to the https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/parts/preprocessing/features.py .

**Comment by blisc** [comment]
Have you tried using our audio loading classes rather than torch audio? It could just be because the audio is getting normalized in torchaudio

**Comment by shangeth** [comment]
Hi @blisc ,
I tried 
- torchaudio.load(audio_path, normalize=False)
- librosa.load()

Still the same result. I couldn't find the audio loading classes of Nemo, can you point me to the file?

**Comment by shangeth** [comment]
You can check/run [this google colab notebook](https://colab.research.google.com/drive/1L-ebrXyruob7xzuVF0fIHN1ajfouOUUS?usp=sharing)

**Comment by blisc** [comment]
Audio_len is in samples not in seconds
`audio_len = torch.tensor(audio.size(1)/22050).view(1,1)` should be `audio_len = torch.tensor(audio.size(1)).view(1,1)`

**Comment by shangeth** [comment]
Thanks a lot, @blisc .
 Yes, that was the mistake. I understand the code base is huge, but it would be nice if there is documentation for utils like audio_to_melspec_precessor/..... 

Closing this issue, Thanks for your help.
